packages/api-graphql/src/internals/v6.ts

```
@@ -106,6 +106,7 @@ export function graphql<
 	const internals = getInternals(this as any);
 	options.authMode = options.authMode || internals.authMode;
 	options.authToken = options.authToken || internals.authToken;
+	const headers = additionalHeaders || internals.headers;
```

This fixes an existing bug - we don't actually pass headers defined in `generateClient` through to `graphql` today.

---

We have tests in the `amplify-api-next` defined behavior tests for client-level headers on modeled operations in `2-expected-use/auth-modes.ts`.

Is the gap *just* for the raw `graphql()` operation? Or, are those modeled operation tests silently failing?

---

Maybe I'm confused, actually. What I thought you were saying seems to be covered in the core `graphql()` case also. We have a test [here](https://github.com/aws-amplify/amplify-js/blob/main/packages/api-graphql/__tests__/GraphQLAPI.test.ts#L1206) that appears to cover headers supplies at various levels as well.

---

1. Added here - https://github.com/aws-amplify/amplify-js/pull/13735/commits/3668cdfbd1336a4dfe114c808bc4f2450cd62b80
2. For posterity - modeled operations perform an additional step where they extract the headers from the client, then pass them through explicitly via the 2nd param to `client.graphql` (example for [list](https://github.com/aws-amplify/amplify-api-next/blob/main/packages/data-schema/src/runtime/internals/operations/list.ts#L94)). So they don't rely on the client-bound headers being passed through automatically, i.e. the patch introduced in this PR.